src/modules/markets/components/markets-view.jsx

```diff
@@ -1,43 +1,156 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet';
 
-import MarketsHeaders from 'modules/markets/components/markets-headers';
+import MarketsHeader from 'modules/markets/components/markets-header';
 import MarketsList from 'modules/markets/components/markets-list';
 import Branch from 'modules/branch/components/branch';
 
-const MarketsView = p => (
-  <section id="markets_view">
-    {!!p.loginAccount.rep && !!p.loginAccount.rep.value && !!p.branch.id &&
-      <Branch {...p.branch} />
+import getValue from 'utils/get-value';
+import parseQuery from 'modules/app/helpers/parse-query';
+import isEqual from 'lodash/isEqual';
+
+import parsePath from 'modules/app/helpers/parse-path';
+import makePath from 'modules/app/helpers/make-path';
+
+import { FAVORITES, MARKETS } from 'modules/app/constants/views';
+import { TOPIC_PARAM_NAME } from 'modules/app/constants/param-names';
+
+export default class MarketsView extends Component {
```

Comment:
I think I'm repeating myself here, but, `MarketsView` is substantially more complex than before.  Is this solely because state is now being held in the component instead of in Redux, or are there other functional changes buried in here?